collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1132,6 +1132,16 @@ public void onCreateContextMenu(ContextMenu menu, View v,
     @Override
     public boolean onContextItemSelected(MenuItem item) {
         if (item.getItemId() == DELETE_REPEAT) {
+            formEntryViewModel.setOnDeleted(() -> {
```

Comment:
I don't think this works because the `ViewModel` will have a reference to the `Activity`, causing a leak.

I thought the callbacks you had before were fine. I was imagining the `ViewModel` only as a way for the dialog to get access to the `FormController`. Even better would be if it gave access only to the data that the dialog needs which I think is the last deleted repeat name and the last deleted repeat index.

Reply:
Sorry @ln I didn't realize this would cause activity leak.